### ISSUE TITLE
SOLVED: Import Loader from skyfield.api crashes with numpy >= 2.0.0

### ISSUE THREAD

**Issue by aendie** [skyfielders/python-skyfield#1005]
I detected that the following statement crashes if numpy 2.0.0 or higher is installed:
```
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

from skyfield.api import Loader
```
Here is the terminal output with Skyfield 1.47 installed:

![image](https://github.com/user-attachments/assets/3cf03135-a7e9-4454-bebf-3a969883465c)

Now downgrade numpy to the highest version before 2.0.0 and it works flawlessly:

![image](https://github.com/user-attachments/assets/77927d28-2d59-4b5a-bb8e-42628a9d8d0b)

**This problem is avoided if Skyfield 1.48 or higher is installed**, however this is not obvious as no mention of it is made in the changelog.

Without extensive testing it appears that all Skyfield versions prior to 1.48 exhibit this problem with numpy >= 2.0.0

**Comment by brandon-rhodes** [comment]
Thanks for verifying that compatibility with NumPy 2 was indeed achieved! I've updated the changelog, which should update on the website the next time I release Skyfield.

**Comment by aendie** [comment]
For informational purposes...
I would like to add here a crash THAT IS NOT DUE TO Skyfield, although it initially appears so.
It is actually an incompatibility between numpy 2.0.0 (and higher) with older versions of pandas (less than 2.2.2). All you need is this code:
```
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

from skyfield.api import load
from skyfield.data import hipparcos

# this crashes with numpy >= 2.0.0 and pandas < 2.2.2
with load.open(hipparcos.URL) as f:
    df = hipparcos.load_dataframe(f)
```
Execute the following to reproduce the crash:
```
pip install numpy --upgrade
pip install pandas==1.5.2
```
and execute the program above. You will see this:
```
Traceback (most recent call last):
  File "D:\_DEVelopment\Astronomical coding\Z_STUFF\z_error caused by numpy 2.0\test_v1.py", line 9, in <module>
    df = hipparcos.load_dataframe(f)
  File "C:\Python310\lib\site-packages\skyfield\data\hipparcos.py", line 54, in load_dataframe
    from pandas import read_csv
  File "C:\Python310\lib\site-packages\pandas\__init__.py", line 22, in <module>
    from pandas.compat import is_numpy_dev as _is_numpy_dev  # pyright: ignore # noqa:F401
  File "C:\Python310\lib\site-packages\pandas\compat\__init__.py", line 18, in <module>
    from pandas.compat.numpy import (
  File "C:\Python310\lib\site-packages\pandas\compat\numpy\__init__.py", line 4, in <module>
    from pandas.util.version import Version
  File "C:\Python310\lib\site-packages\pandas\util\__init__.py", line 2, in <module>
    from pandas.util._decorators import (  # noqa:F401
  File "C:\Python310\lib\site-packages\pandas\util\_decorators.py", line 14, in <module>
    from pandas._libs.properties import cache_readonly
  File "C:\Python310\lib\site-packages\pandas\_libs\__init__.py", line 13, in <module>
    from pandas._libs.interval import Interval
  File "pandas\_libs\interval.pyx", line 1, in init pandas._libs.interval
ValueError: numpy.dtype size changed, may indicate binary incompatibility. Expected 96 from C header, got 88 from PyObject
```
which is telling the user it crashed while executing `df = hipparcos.load_dataframe(f)`.

However, this is actually caused by the _pandas_ version being incompatible with _numpy_ 2.0.0 (or higher).
Now execute the following statement:
```
pip install pandas --upgrade
```
and the test program runs without any error.

This is not the fault of Skyfield, so some code can be added to detect the incompatible versions and inform the user with a user-friendly message. As an example, one way to achieve this is to modify the test program as follows:
```
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

def compareVersion(version1, version2):     # compare two versions
    versions1 = [int(v) for v in version1.split(".")]
    versions2 = [int(v) for v in version2.split(".")]
    for i in range(max(len(versions1),len(versions2))):
        v1 = versions1[i] if i < len(versions1) else 0
        v2 = versions2[i] if i < len(versions2) else 0
        if   v1 > v2: return 1
        elif v1 < v2: return -1
    return 0

import sys
n = sys.version.find(" ")
py_ver = sys.version[:n]        # python version

# check if pandas version is compatible with numpy
# (this is required to prevent a crash in "df = hipparcos.load_dataframe(f)")
if compareVersion(py_ver,"3.8") >= 0:
    from importlib.metadata import version          # for Python >= 3.8
    pandas_ver = version('pandas')
    np_ver = version('numpy')

    if compareVersion(np_ver,"2.0.0") >= 0:         # if numpy >= 2.0.0
        if compareVersion(pandas_ver, "2.2.2") < 0: # if pandas < 2.2.2
            print("ISSUE: pandas version {} is incompatible with numpy {}".format(pandas_ver, np_ver))
            print("       upgrade pandas to >= 2.2.2 or downgrade numpy to <= 1.26.4")
            sys.exit(0)
else:   # if Python < 3.8
    import numpy as np
    try:
        import pandas
    except ValueError:
        print("ISSUE: pandas version is incompatible with numpy {}".format(np.__version__))
        print("       upgrade pandas to >= 2.2.2 or downgrade numpy to <= 1.26.4")
        sys.exit(0)

from skyfield.api import load
from skyfield.data import hipparcos

with load.open(hipparcos.URL) as f:
    df = hipparcos.load_dataframe(f)
```
Note that `importlib.metadata` is only available in Python 3.8 onwards. The test code above in case Python is older than version 3.8 is not rigorous, however it will still catch the incompatibility (and is sufficient as few users are likely to be on Python 3.7 or older).
